saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -0,0 +1,1067 @@
+        order_instance.undiscounted_total_net_amount = (
+            order_amounts.undiscounted_total_net
+        )
+        order_instance.customer_note = order_input.get("customer_note", "")
+        order_instance.redirect_url = order_input.get("redirect_url")
+        order_instance.origin = OrderOrigin.BULK_CREATE
+        order_instance.weight = order_input.get("weight", zero_weight())
+        order_instance.tracking_client_id = order_input.get("tracking_client_id")
+        order_instance.currency = order_input["currency"]
+        order_instance.should_refresh_prices = False
+        update_order_display_gross_prices(order_instance)
+
+        if metadata := delivery_method.shipping_tax_class_metadata:
+            for data in metadata:
+                order_instance.shipping_tax_class_metadata.update(
+                    {data["key"]: data["value"]}
+                )
+        if private_metadata := delivery_method.shipping_tax_class_private_metadata:
+            for data in private_metadata:
+                order_instance.shipping_tax_class_private_metadata.update(
+                    {data["key"]: data["value"]}
+                )
+
+        # TODO charged
+        # TODO authourized
+        # TODO voucher
+        # TODO gift cards
+
+        order.order = order_instance
+        return order
+
+    @classmethod
+    def handle_error_policy(
+        cls,
+        orders: List[OrderWithErrors],
+        policy: ErrorPolicy,
+    ):
+        errors = [error for order in orders for error in order.get_all_errors()]
+        if errors:
+            for order in orders:
+                if policy == ErrorPolicy.REJECT_EVERYTHING:
+                    order.order = None
+                elif policy == ErrorPolicy.REJECT_FAILED_ROWS:
+                    if order.get_all_errors():
+                        order.order = None
+        return orders
+
+    @classmethod
+    @traced_atomic_transaction()
+    def save_data(cls, orders: List[OrderWithErrors]):
+        addresses = []
+        for order in orders:
+            if order.order:
+                if billing_address := order.order.billing_address:
+                    addresses.append(billing_address)
+                if shipping_address := order.order.shipping_address:
+                    addresses.append(shipping_address)
+        Address.objects.bulk_create(addresses)
+
+        Order.objects.bulk_create([order.order for order in orders if order.order])
+
+        order_lines = [line for order in orders for line in order.lines if order.order]
+        OrderLine.objects.bulk_create(order_lines)
+
+        notes = [note for order in orders for note in order.notes if order.order]
+        OrderEvent.objects.bulk_create(notes)
+
+        return orders
+
+    @classmethod
+    def perform_mutation(cls, _root, _info: ResolveInfo, /, **data):
+        # TODO post save actions
+        # TODO add webhook ORDER_BULK_CREATED
```

Comment:
What action should be taken regarding this TODO?

Reply:
This will be handled in separate PRs - description updated.